datafusion/expr/src/expr_fn.rs

```diff
@@ -103,6 +105,50 @@ pub fn or(left: Expr, right: Expr) -> Expr {
     ))
 }
 
+// There is a solid implementation elsewhere. We are going to wrap it up as a function.
+macro_rules! naive_unary_fn_impl {
+    ($func:ident, $doc:expr) => {
+        #[doc = $doc]
+        pub fn $func(expr: Expr) -> Expr {
+            expr.$func()
+        }
+    };
+}
+
+macro_rules! naive_binary_fn_impl {
+    ($func:ident, $doc:expr) => {
+        #[doc = $doc]
+        pub fn $func(left: Expr, right: Expr) -> Expr {
+            left.$func(right)
+        }
+    };
+    ($func:ident, $from_func:ident, $doc:expr) => {
+        #[doc = $doc]
+        pub fn $func(left: Expr, right: Expr) -> Expr {
+            left.$from_func(right)
+        }
+    };
+}
+
+// Arithmetic expression
+naive_binary_fn_impl!(add, "Arithmetic addition");
+naive_binary_fn_impl!(sub, "Arithmetic subtraction");
+naive_binary_fn_impl!(mul, "Arithmetic multiplication");
+naive_binary_fn_impl!(div, "Arithmetic division");
+naive_binary_fn_impl!(r#mod, r#mod, "Arithmetic remainder");
```

Comment:
is it a typo? shouldn't be it 
```
naive_binary_fn_impl!(r#mod, rem, "Arithmetic remainder");
```

Reply:
Yes, you're right.
I'm intended to do that at first, but since `Expr::r#mod` also implemented in this PR, this can be
```
naive_binary_fn_impl!(r#mod, "Arithmetic remainder");
```